### PR TITLE
8307653: Adjust delay time and gc log argument in TestAbortOnVMOperationTimeout

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch increases the delay time of the test `TestAbortOnVMOperationTimeout` 
so that it can pass on the low performance devices (such as my local riscv64 dev board).

And I adjust the gc log options to get more gc log message which is useful to
inspect the issue [JDK-8296819](https://bugs.openjdk.org/browse/JDK-8296819).

Thanks for the review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307653](https://bugs.openjdk.org/browse/JDK-8307653): Adjust delay time and gc log argument in TestAbortOnVMOperationTimeout

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13878/head:pull/13878` \
`$ git checkout pull/13878`

Update a local copy of the PR: \
`$ git checkout pull/13878` \
`$ git pull https://git.openjdk.org/jdk.git pull/13878/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13878`

View PR using the GUI difftool: \
`$ git pr show -t 13878`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13878.diff">https://git.openjdk.org/jdk/pull/13878.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13878#issuecomment-1539440887)

### CODE REVIEW

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java [hunk 1]
```diff
@@ -51,9 +51,9 @@ public static void main(String[] args) throws Exception {
             return;
         }
 
-        // These should definitely pass: more than a minute is enough for Serial to act.
+        // These should definitely pass: more than 3 minutes is enough for Serial to act.
         // The values are deliberately non-round to trip off periodic task granularity.
-        for (int delay : new int[]{63423, 12388131}) {
+        for (int delay : new int[]{183423}) {
             testWith(delay, true);
         }
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:58]: With only one value there is no point having the loop. Why not just replace the shorter, failing, delay with the new longer one?

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java (hunk 1 above)
Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:58]: > With only one value there is no point having the loop

Good caught. I just obey the previous code style and forget to refactor it. Will fix it.

> Why not just replace the shorter, failing, delay with the new longer one?

The longer one is useless actually. If the shorter one, `63423` previously, passed,
it means the gc time is less than `63423`. Then the longer one will always pass.
And jtreg would report `time out` in seraval minutes normally,
so the longer time `12388131` is useless.
So I want to remove the longer one and just remain the shorter one.

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java [hunk 2]
```diff
@@ -26,13 +26,13 @@
...
  * @test TestAbortOnVMOperationTimeout
- * @bug 8181143 8269523
+ * @bug 8181143 8269523 8307653
  * @summary Check abort on VM timeout is working
  * @requires vm.flagless
  * @library /test/lib
  * @modules java.base/jdk.internal.misc
  *          java.management
- * @run driver TestAbortOnVMOperationTimeout
+ * @run driver/timeout=600 TestAbortOnVMOperationTimeout
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: Is this change really needed? Do you see timeouts without it?

Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: Yes, local env shows `time out`. The jtreg's default timeout is `480s`, so I increase it to `600s`.

Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: Not quite. The default JTREG timeout is 120 seconds and the default timeoutFactor
value that we use is 4 for a total timeout value of 480 seconds. If your test runs
indicate the need for a total timeout value of 600 seconds, then you should use:
```
 * @run driver/timeout=150
```

Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: I had tried `@run driver/timeout=480`, but the test failed. Then I used `timeout=600`, the test passed. So I set it to `600`.

Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: What is your environment's timeoutFactor value?
Also, how are you running the test?

Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: Maybe I should ignore such `time out` tests in my low performance dev board and don't fix it?

But such `time out` indeed pullutes the test result, and if I don't fix it,
I always need to read the failed test result to judge which tests fail.
In other words, my local CI will never be green.

Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: When I run the test via:

Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: > What is your environment's timeoutFactor value?

I haven't changed the default value. So it should be 4.

> Also, how are you running the test?

I used the command `make test Test=runtime/Safepoint/TestAbortOnVMOperationTimeout.java`.

Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: Please search your .jtr file for your run for the pattern: "test.timeout.factor"
and report the results here.

Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: I see the text `-J-Dtest.timeout.factor=4.0` in the log.

Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: When you get a timeout with 'timeout=480', what 'elapsed time' values are in the '.jtr' file.
For my 'fastdebug' run, I see:
```
$ grep 'elapsed time' build/macosx-x86_64-normal-server-fastdebug/test-support/jtreg_open_test_hotspot_jtreg_runtime_Safepoint_TestAbortOnVMOperationTimeout_java/runtime/Safepoint/TestAbortOnVMOperationTimeout.jtr
elapsed time (seconds): 3.458
elapsed time (seconds): 3.455
elapsed time (seconds): 10.176
```

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java [hunk 3]
```diff
@@ -51,11 +51,9 @@ public static void main(String[] args) throws Exception {
             return;
         }
 
-        // These should definitely pass: more than a minute is enough for Serial to act.
+        // These should definitely pass: more than 3 minutes is enough for Serial to act.
```
Comment (dcubed-ojdk) [inline test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:54]: "These" is no longer accurate since this has been changed to test a single value.

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java [hunk 4]
```diff
@@ -51,11 +51,9 @@ public static void main(String[] args) throws Exception {
...
         // The values are deliberately non-round to trip off periodic task granularity.
-        for (int delay : new int[]{63423, 12388131}) {
-            testWith(delay, true);
-        }
+        testWith(183423, true);
```
Comment (dcubed-ojdk) [inline test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:56]: "The values are ..." is no longer accurate since this has been changed to test a single value.

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

**Comment by lgxbslgx** [comment]
> The proper way to address a slow piece of hardware is to use a larger timeoutFactor value.

Got it. I will remove the `/timeout=600` of the test.

**Comment by dcubed-ojdk** [comment]
BTW, my first experiment for [JDK-8296819](https://bugs.openjdk.org/browse/JDK-8296819) would
have been switching from (default) timeout=120 to timeout=180 and running that thru the stress
kit in my lab.

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java (hunk 2 above)
Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java:35]: I can't reproduce the failed test by using `@run driver/timeout=480` now. The test passes locally now with `timeout=480`. The `elapsed time` values, which belong to the current passed test, are shown below.

```
elapsed time (seconds): 115.718
elapsed time (seconds): 115.699
elapsed time (seconds): 555.602
```

I try to recall my previous steps:
- The test failed locally with `@run driver`,  without `timeout` option.
- I changed it to `@run driver/timeout=900`. It passed.
- I changed it to `@run driver/timeout=480`. It failed.
- I changed it to `@run driver/timeout=600`. It passed.

So I push the code with `@run driver/timeout=600`.

Now I try `@run driver/timeout=480` locally, it can't reproduce. So strange! Such intermittent fail may be related to [JDK-8296819](https://bugs.openjdk.org/browse/JDK-8296819)?

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java (hunk 3 above)
Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java]: Fixed.

File: test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java (hunk 4 above)
Comment (lgxbslgx) [reply test/hotspot/jtreg/runtime/Safepoint/TestAbortOnVMOperationTimeout.java]: Fixed.

**Review by dholmes-ora — APPROVED** [review]
I am okay with these changes.

I do worry that your slow dev board may expose a lot of internal timeout issues in tests - but tests should be using values modified by the timeoutFactor to compensate for slow systems where possible. Not sure how that would apply in the current case though.

**Review by dcubed-ojdk — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@dholmes-ora @dcubed-ojdk Thanks for the review. Integrating.
/integrate
